packages/peregrine/lib/talons/Header/useCartTrigger.js

```diff
@@ -1,32 +1,46 @@
 import { useCallback, useEffect } from 'react';
-import { useMutation } from '@apollo/react-hooks';
-import { useCartContext } from '@magento/peregrine/lib/context/cart';
+import { useLazyQuery } from '@apollo/react-hooks';
+
 import { useAppContext } from '@magento/peregrine/lib/context/app';
-import { useAwaitQuery } from '@magento/peregrine/lib/hooks/useAwaitQuery';
+import { useCartContext } from '@magento/peregrine/lib/context/cart';
 
-export const useCartTrigger = props => {
-    const { createCartMutation, getCartDetailsQuery } = props;
-    const [, { toggleDrawer }] = useAppContext();
-    const [{ derivedDetails }, { getCartDetails }] = useCartContext();
-    const { numItems: itemCount } = derivedDetails;
+import gql from 'graphql-tag';
 
-    const [fetchCartId] = useMutation(createCartMutation);
-    const fetchCartDetails = useAwaitQuery(getCartDetailsQuery);
+const GET_ITEM_QUANTITY_GQL = gql`
+    query getCartDetails($cartId: String!) {
+        cart(cart_id: $cartId) {
+            items {
+                id
+                quantity
+            }
+        }
+    }
+`;
+
+const getTotalQuantity = items =>
+    items.reduce((total, item) => total + item.quantity, 0);
+
+export const useCartTrigger = () => {
+    const [, { toggleDrawer }] = useAppContext();
+    const [{ cartId }] = useCartContext();
+    const [getItemQuantities, { data }] = useLazyQuery(GET_ITEM_QUANTITY_GQL);
 
     useEffect(() => {
-        getCartDetails({
-            fetchCartId,
-            fetchCartDetails
-        });
-    }, [fetchCartDetails, fetchCartId, getCartDetails]);
+        if (cartId) {
+            getItemQuantities({
```

Comment:
This is only called once! After that, each time the cart is updated (as in adding/removing a product) the data from the lazy query is updated as well!